src/libs/API/types.ts

```diff
@@ -0,0 +1,312 @@
+    [READ_COMMANDS.OPEN_PERSONAL_DETAILS_PAGE]: EmptyObject;
+    [READ_COMMANDS.OPEN_PUBLIC_PROFILE_PAGE]: Parameters.OpenPublicProfilePageParams;
+    [READ_COMMANDS.OPEN_PLAID_BANK_LOGIN]: Parameters.OpenPlaidBankLoginParams;
+    [READ_COMMANDS.OPEN_PLAID_BANK_ACCOUNT_SELECTOR]: Parameters.OpenPlaidBankAccountSelectorParams;
+    [READ_COMMANDS.GET_OLDER_ACTIONS]: Parameters.GetOlderActionsParams;
+    [READ_COMMANDS.GET_NEWER_ACTIONS]: Parameters.GetNewerActionsParams;
+    [READ_COMMANDS.EXPAND_URL_PREVIEW]: Parameters.ExpandURLPreviewParams;
+    [READ_COMMANDS.GET_REPORT_PRIVATE_NOTE]: Parameters.GetReportPrivateNoteParams;
+    [READ_COMMANDS.OPEN_ROOM_MEMBERS_PAGE]: Parameters.OpenRoomMembersPageParams;
+    [READ_COMMANDS.SEARCH_FOR_REPORTS]: Parameters.SearchForReportsParams;
+    [READ_COMMANDS.SEND_PERFORMANCE_TIMING]: Parameters.SendPerformanceTimingParams;
+    [READ_COMMANDS.GET_ROUTE]: Parameters.GetRouteParams;
+    [READ_COMMANDS.GET_ROUTE_FOR_DRAFT]: Parameters.GetRouteForDraftParams;
+    [READ_COMMANDS.GET_STATEMENT_PDF]: Parameters.GetStatementPDFParams;
+    [READ_COMMANDS.OPEN_ONFIDO_FLOW]: EmptyObject;
+    [READ_COMMANDS.OPEN_INITIAL_SETTINGS_PAGE]: EmptyObject;
+    [READ_COMMANDS.OPEN_ENABLE_PAYMENTS_PAGE]: EmptyObject;
+    [READ_COMMANDS.BEGIN_SIGNIN]: Parameters.BeginSignInParams;
+    [READ_COMMANDS.SIGN_IN_WITH_SHORT_LIVED_AUTH_TOKEN]: Parameters.SignInWithShortLivedAuthTokenParams;
+    [READ_COMMANDS.OPEN_WORKSPACE_REIMBURSE_VIEW]: Parameters.OpenWorkspaceReimburseViewParams;
+    [READ_COMMANDS.OPEN_WORKSPACE]: Parameters.OpenWorkspaceParams;
+    [READ_COMMANDS.OPEN_WORKSPACE_MEMBERS_PAGE]: Parameters.OpenWorkspaceMembersPageParams;
+    [READ_COMMANDS.OPEN_WORKSPACE_INVITE_PAGE]: Parameters.OpenWorkspaceInvitePageParams;
+    [READ_COMMANDS.OPEN_DRAFT_WORKSPACE_REQUEST]: Parameters.OpenDraftWorkspaceRequestParams;
+};
+
+const SIDE_EFFECT_REQUEST_COMMANDS = {
+    AUTHENTICATE_PUSHER: 'AuthenticatePusher',
+    OPEN_REPORT: 'OpenReport',
+    OPEN_OLD_DOT_LINK: 'OpenOldDotLink',
+    REVEAL_EXPENSIFY_CARD_DETAILS: 'RevealExpensifyCardDetails',
+    GET_MISSING_ONYX_MESSAGES: 'GetMissingOnyxMessages',
+    RECONNECT_APP: 'ReconnectApp',
+} as const;
+
+type SideEffectRequestCommand = ReadCommand | ValueOf<typeof SIDE_EFFECT_REQUEST_COMMANDS>;
+
+type SideEffectRequestCommandParameters = ReadCommandParameters & {
+    [SIDE_EFFECT_REQUEST_COMMANDS.AUTHENTICATE_PUSHER]: Parameters.AuthenticatePusherParams;
+    [SIDE_EFFECT_REQUEST_COMMANDS.OPEN_REPORT]: Parameters.OpenReportParams;
+    [SIDE_EFFECT_REQUEST_COMMANDS.OPEN_OLD_DOT_LINK]: Parameters.OpenOldDotLinkParams;
+    [SIDE_EFFECT_REQUEST_COMMANDS.REVEAL_EXPENSIFY_CARD_DETAILS]: Parameters.RevealExpensifyCardDetailsParams;
+    [SIDE_EFFECT_REQUEST_COMMANDS.GET_MISSING_ONYX_MESSAGES]: Parameters.GetMissingOnyxMessagesParams;
+    [SIDE_EFFECT_REQUEST_COMMANDS.RECONNECT_APP]: Parameters.ReconnectAppParams;
+};
```

Comment:
Instead of extending the side effects commands to include the read commands. Extend the `makeRequestWithSideEffects` function to support read and write commands

Reply:
Same thing [here](https://github.com/Expensify/App/pull/34542#discussion_r1464729790)

Reply:
I don't think that's the same point. What I meant here is that the `SideEffectRequestCommandParameters` type should not include `ReadCommandParameters`.